week1/homework/src/server.js

```diff
@@ -9,7 +8,35 @@ function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    response.setHeader('Content-Type', 'application/json', 'charset = utf - 8');
+    switch (request.url) {
+      case '/state':
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/add':
+        state += 1;
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/subtract':
+        state -= 1;
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/reset':
+        state = 10;
+        response.write(JSON.stringify({ state }));
+        break;
+      default:
+        if (request.url === '/') {
```

Comment:
> Any other URL results in an error with status code 404 and response body in JSON format

https://github.com/HackYourFuture/Node.js/blob/master/week1/homework/README.md#endpoints-to-implement

So I think root also should yield to 404

But in case you want to accept calls to '/' then you can combine the two cases ('/' and '/state') together

    switch (request.url) {
      case '/':
      case '/state':
        response.write(JSON.stringify({ state }));
        break;
      // ... the rest

Ref. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#Multi-case_-_single_operation